crates/ruff/src/rules/dlint/rules/bad_compile_use.rs

```diff
@@ -0,0 +1,30 @@
+use rustpython_parser::ast;
+use rustpython_parser::ast::{Expr, Ranged};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+#[violation]
+pub struct BadCompileUse;
+
+impl Violation for BadCompileUse {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use of the `compile` command should be avoided")
+    }
+}
+
+/// DUO110
+pub(crate) fn bad_compile_use(checker: &mut Checker, expr: &Expr) {
+    if let Expr::Call(ast::ExprCall { func, .. }) = expr {
+        if let Some(call_path) = checker.ctx.resolve_call_path(func) {
```

Comment:
Nit: It could make sense to first test if the function called by `func` matches the name "compile" to avoid the somewhat expensive `resolve_call_path` call.

Do I understand correctly that the rule is about the built-in `compile` function? If so, you can use checker.ctx.is_builtin(name)` to check if the name is not a user-defined function.

Reply:
It is, so I'll use `is_builtin` method. But good to know, I'll open a PR soon with a micro optimization for my previous implementations of similar funcs and keep it in mind for future changes!